libfwupdplugin/fu-sbatlevel-section.c

```diff
@@ -52,16 +53,15 @@ fu_sbatlevel_section_add_entry(FuFirmware *firmware,
 	fu_firmware_set_idx(entry_fw, entry_idx);
 	fu_firmware_set_id(entry_fw, entry_name);
 	fu_firmware_set_offset(entry_fw, offset);
-	entry_blob = fu_bytes_new_offset(fw, offset, size, error);
-	if (entry_blob == NULL)
-		return FALSE;
-	if (!fu_firmware_add_image_full(firmware, entry_fw, error))
-		return FALSE;
-	if (!fu_firmware_parse(entry_fw, entry_blob, flags, error)) {
+	partial_stream = fu_partial_input_stream_new(stream, offset, streamsz - offset);
+	if (!fu_firmware_parse_stream(entry_fw, partial_stream, 0, flags, error)) {
```

Comment:
I suppose we just rely on `fu_firmware_parse_stream` doing `g_return_val_if_fail(G_IS_INPUT_STREAM(stream), FALSE)` to handle the rare case where `fu_partial_input_stream_new` returned `NULL`. I was wondering if it is normal we do not check the return of `fu_partial_input_stream_new`.

Reply:
Not sure I follow; why would `fu_partial_input_stream_new()` return with failure?